Northeastern-Electric-Racing/FinishLine
# #956 - Remove Task List Loading Indicator With No Team

## Changes

Instead of displaying loading symbol if a project has no team, display helpful text message instead.

New Task button also disabled if no team. Task edit permission is controlled *per task* now.

Refactor task list test data using mock hooks, and change task list props to pass the whole project. Plus some free task test data I didn't get to use.

## Notes

I chose to display text instead of nothing to make it more user-friendly.

Did not do similar changes for `!auth.user` (will that ever happen?) or `AssigneeEdit` component (does this ever get displayed?)

## Test Cases

- New Task Button
  - Disabled if no team
  - (manual) Assigning a project to a team will instantly refresh component to show table and enable button
- Displaying "no team" message
  - Renders if no associated team
  - No render if there is a team

Manually checking edit permission is controlled per task:
1. Use admin tools to demote Joe Shmoe to guest.
2. Log in as Joe Shmoe
3. Project 1.1.0 > tasks In Progress
4. "Decorate Impact Attenuator" should be enabled
5. "Meet with the Department of Transportation" should be disabled (both in move/delete task menu and in the edit modal)

## Screenshots

![Screen Shot 2023-03-07 at 9 54 06 PM](https://user-images.githubusercontent.com/59806366/223608120-ebe97b70-dcbc-450b-9504-0679d46711ad.png)
![Screen Shot 2023-03-07 at 9 54 52 PM](https://user-images.githubusercontent.com/59806366/223608116-2554223a-7ec1-4688-a557-4a42fe6eec8e.png)
![Screen Shot 2023-03-07 at 9 54 13 PM](https://user-images.githubusercontent.com/59806366/223608119-1e874e90-c486-46ae-b1ef-8574123b1930.png)

## To Do

_Any remaining things that need to get done_

- [x] Disable New Task button if there is no team. (Is there some way to couple all this together?)

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.


Closes #956
